src/data-workspace/validation/validation-results-sidebar.js

```diff
@@ -27,15 +27,31 @@ export default function ValidationResultsSidebar() {
             (ruleViolations) => ruleViolations.length === 0
         )
 
-    const hasCommentsViolations = commentRequiredViolations?.length
+    const hasCommentsViolations = Boolean(commentRequiredViolations?.length)
 
+    const rerunValidation = () => {
+        rightHandPanel.setFormChangedSincePanelOpened(false)
+        refetch()
+    }
```

Comment:
is that to optimise performance? If yes, I think the cost of inline functions is quite minimal that it's considered less performant to useUseCallBack than to to just have the inline function, so unless we're passing that function as prop, it's better to leave it here .. here is [a blog from Kent Dodds](https://kentcdodds.com/blog/usememo-and-usecallback) about it, and an [SO post](https://stackoverflow.com/a/57156669).

Reply:
No, not to optimize performance at all but to ensure that code doesn't have to worry about stable references

Reply:
why are stable references an issue (other than in performance context)? these methods are re-created in functional components, and that's transparent from other parts of the code from what I understand

Reply:
At the moment there's no issue. An issue could pop up when adding a callback to a dependency array of another hook, which can cause buggy behavior, and these are generally quite hard to track. It's something I encountered a couple of times when doing the performance work, but I also had to debug some nasty bugs in the past where a ref wasn't stable and the app ended up in infinite loops or simply did weird things.

We can leave it as is and leave it up to you if you want to wrap this wtih a `useCallback` or not.
I have had a couple of issues with this in the past, simply because it wasn't a problem at first but later, some other part of the code expected the reference to be stable. Again, this is not about performance. I'm generally against premature performance optimizations, but for making code more resilient. I don't think that the performance cost of a `useCallback` is significant in this case, the hook would add a guard against hard-to-debug bugs on the other hand. At least right now I don't see a downside of using `useCallback`

Reply:
I guess you're right. It's not a good idea to simply wrap every function with `useCallback`, I probably got traumatized when working on the performance optimization..

Reply:
ok, I will leave it for now but maybe we can have a wider discussion on _defaulting_ to use `useCallback`. My last project actually had a rule to always use it, but that was seen as a performance optimisation for react-native. In my head having a memoised reference _could_ cause hard to debug issues, but that's just a theoretical assumption - I haven't seen that happening in real life ..  so it would be good to have a discussion and see examples of the bugs that could happen.